src/pgwire/src/protocol.rs

```diff
@@ -911,6 +911,34 @@ where
         Ok(State::Ready)
     }
 
+    async fn send<M>(&mut self, message: M) -> Result<(), io::Error>
+    where
+        M: Into<BackendMessage>,
+    {
+        let message: BackendMessage = message.into();
+        let minimum_client_severity = self.coord_client.session().vars().client_min_messages();
```

Comment:
Maybe this should only be fetched if we have a error message to avoid needlessly calling three functions for each message.